tuxemon/event/actions/overwrite_tech.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2024 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+from __future__ import annotations
+
+import logging
+import uuid
+from dataclasses import dataclass
+from typing import final
+
+from tuxemon.event.eventaction import EventAction
+from tuxemon.monster import Monster
+from tuxemon.technique.technique import Technique
+
+logger = logging.getLogger(__name__)
+
+
+@final
+@dataclass
+class OverwriteTechAction(EventAction):
+    """
+    Overwrite / replace a technique with another.
+
+    Script usage:
+        .. code-block::
+
+            overwrite_tech <removed>,<added>
+
+    Script parameters:
+        removed: Name of the variable where to store the tech id.
+        added: Slug technique.
+
+    eg. "overwrite_tech name_variable,peck"
+
+    """
+
+    name = "overwrite_tech"
+    removed: str
+    added: str
+
+    def overwrite(self, monster: Monster, removed: Technique) -> None:
+        slot = monster.moves.index(removed)
```

Comment:
the logic here is fine, but im wondering why you you decided not to use the list indexes
```python
slot = monster.moves.index(removed)
monster.moves[slot] = added
```
not worth changing, but im just curious